### PR TITLE
[feature] Method argument multi line (wip)

### PR DESCRIPTION
**Pull request by SpacePossum**
This PR fixes a few issues with the `MethodArgumentSpaceFixer`.

It removes white space between the method opening bracket and the first argument,
and between the last argument and the method closing round bracket.
For example:

```php
<?php xyz(   "", "" ,"" ,""  );
```
will now be fixed to:
```php
<?php xyz("", "", "", "");
```

It will add line breaks after the arguments if there is one line break found between the arguments.
It does not fix the indenting of these new lines (which reminds me to add a fixer order test).

For example:
```php
<?php function("e","f", // comment 1
                "g","h"  ) /* comment 2 */;
```
To:
```php
<?php function(
"e",
"f", // comment 1
                "g",
"h"
) /* comment 2 */;
```

I also made a method of the fixer private, which I don't consider a BC break since the Fixer is already final and internal.

see also:
https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-2-coding-style-guide.md#44-method-arguments

https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-2-coding-style-guide.md#46-method-and-function-calls

and https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/1397
There a few cases where the current codebase is not following the PSR, so I expect the tests to fail.

I'll work on this a bit, like the missing order test, a 1.x version and the CS fixes of the project itself, if this seems a good idea ^-^

### CODE REVIEW

**Comment by gharlan** [comment]
> It removes white space between the method opening bracket and the first argument,
> and between the last argument and the method closing round bracket.

Isn't it the job of [ParenthesisFixer](https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/1.11/Symfony/CS/Fixer/PSR2/ParenthesisFixer.php)?

**Comment by keradus** [comment]
Why targetting master and not 1.11 ?

**Comment by SpacePossum** [comment]
>>It removes white space between the method opening bracket and the first argument,
and between the last argument and the method closing round bracket.

>Isn't it the job of ParenthesisFixer?

I think the `MethodArgumentSpaceFixer` should fix all spaces around the arguments, not just the middle ones. 

>Why targetting master and not 1.11 ?

I started with the master version because it already has the fixes for the multi line comments, something that is not in 1.11.

**Comment by keradus** [comment]
Does multiline comments needed for multiline args?

File: Symfony/CS/Tests/Fixer/PSR2/MethodArgumentSpaceFixerTest.php [hunk 1]
```diff
@@ -33,8 +33,34 @@ public function testFixProvider()
...
+
+            ),
+            array(
+                '<?php function(
+"e",
+"f", // comment 1
+                "g",
+"h"
+) /* comment 2 */;',
+                '<?php function("e","f", // comment 1
```
Comment (gharlan) [inline Symfony/CS/Tests/Fixer/PSR2/MethodArgumentSpaceFixerTest.php]: this is invalid code (invalid function name).

**Comment by gharlan** [comment]
Is this style supported?

```php
somefunction($foo, $bar, [
  // ...
], $baz);

$app->get('/hello/{name}', function ($name) use ($app) { 
    return 'Hello '.$app->escape($name); 
});
```

https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-2-coding-style-guide-meta.md#31---multi-line-arguments-09082013

File: Symfony/CS/Tests/Fixer/PSR2/MethodArgumentSpaceFixerTest.php [hunk 2]
```diff
@@ -208,16 +228,16 @@ function xyz(
...
         ,
-        <<<'EOTXTb'
+<<<'EOTXTb'
     nowdoc
 EOTXTb
         ,
-        'foo'
-    );
+'foo'
+);
```
Comment (gharlan) [inline Symfony/CS/Tests/Fixer/PSR2/MethodArgumentSpaceFixerTest.php]: why did you change the indentation here?

Comment (GrahamCampbell) [inline Symfony/CS/Tests/Fixer/PSR2/MethodArgumentSpaceFixerTest.php]: to make the tests pass I assume, lol

Comment (GrahamCampbell) [inline Symfony/CS/Tests/Fixer/PSR2/MethodArgumentSpaceFixerTest.php]: this change does need reverting though, and the implementation fixing

Comment (SpacePossum) [inline Symfony/CS/Tests/Fixer/PSR2/MethodArgumentSpaceFixerTest.php]: >why did you change the indentation here?
I forgot to revert this, thanks :+1: 

>to make the tests pass I assume, lol
>this change does need reverting though, and the implementation fixing

no, the test runs just fine with change reverted,
af3ff6258f22d627991a7f04f2ecaef1e9819995

**Comment by SpacePossum** [comment]
>Is this style supported?

No, I was not aware of that part of the PSR :(

File: Symfony/CS/Tests/Fixer/PSR2/MethodArgumentSpaceFixerTest.php (hunk 1 above)
Comment (SpacePossum) [inline Symfony/CS/Tests/Fixer/PSR2/MethodArgumentSpaceFixerTest.php]: thank you, fix in 7a23dbd1b63215fbf87aafae6574f28d784419a9

**Comment by SpacePossum** [comment]
also related: https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/111

**Comment by keradus** [comment]
Any news on that?

**Comment by SpacePossum** [comment]
3. Errata
3.1 - Multi-line Arguments (09/08/2013)

are a bit to much for me at the moment, so I give up on this one for the time being.
